Assets/__Scripts/Beatmap/Base/Customs/INoodleExtensions.cs

```diff
@@ -27,9 +27,23 @@ public interface INoodleExtensionsSlider : INoodleExtensionsGrid
         string CustomKeyTailCoordinate { get; }
     }
 
+    public struct Vector2Or3
+    {
+        public Vector2Or3(float? x, float? y, float? z = null)
+        {
+            this.x = x ?? 0f;
+            this.y = y ?? 0f;
+            this.z = z;
+        }
+
+        public float x { get; }
+        public float y { get; }
+        public float? z { get; }
+    }
```

Comment:
Null z isn't possible with Vector3 so here we are.
Welcome suggestions for where to put this struct as it being in INoodleExtensions seems iffy.

Reply:
i vote `Beatmap/Shared` or `Beatmap/Helper`